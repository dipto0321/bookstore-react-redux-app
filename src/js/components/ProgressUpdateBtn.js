import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { switchProgressUpdate } from '../actions/books';

export const ProgressUpdateBtn = ({ switchProgressUpdate, id }) => (
  <button
    className="update-btn update-btn-primary"
    type="button"
    onClick={() => switchProgressUpdate(id)}
  >
    Update Progress
  </button>
);

ProgressUpdateBtn.propTypes = {
  switchProgressUpdate: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
};

export default connect(null, { switchProgressUpdate })(ProgressUpdateBtn);
