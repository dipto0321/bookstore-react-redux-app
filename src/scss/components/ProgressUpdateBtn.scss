.update-btn {
  width: 184px;
  height: 33px;
  border-radius: 3px;
  font-family: "Roboto Slab", serif;
  font-size: 13px;
  font-weight: 300;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: 0.5px;
  text-align: center;
}
.update-btn-primary {
  color: #ffffff;
  background-color: #0290ff;
  border-color: #0290ff;
}
