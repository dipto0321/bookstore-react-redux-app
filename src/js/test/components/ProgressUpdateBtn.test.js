import React from 'react';
import { shallow } from 'enzyme';
import ProgressUpdateBtn from '../../components/ProgressUpdateBtn';

test('ProgressUpdateBtn should render correctly', () => {
  const wrapper = shallow(<ProgressUpdateBtn />);
  expect(wrapper).toMatchSnapshot();
});
