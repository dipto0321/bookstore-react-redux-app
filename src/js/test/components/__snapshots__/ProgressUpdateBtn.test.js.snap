// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressUpdateBtn should render correctly 1`] = `
<button
  className="update-btn update-btn-primary"
  type="button"
>
  Update Progress
</button>
`;
